### DISCUSSION TITLE
How to solve this set of PDE in GFD?

### DISCUSSION THREAD

**Discussion by miniufo** [zwicker-group/py-pde#432] [Q&A]
Hi, I am recently working on [a PDE solver for geophysical fluid dynamic problems](https://github.com/miniufo/xinvert) and it is really great to come across this excellent package!

Just two quick questions here:

1. Is it possible to use this solver in an arbitary domain like the ocean with arbitary shape of coastlines?
2. I have a problem in hand which is a set of PDEs with two unkown: streamfunction $\psi$ and layer thickness $h$:

$$
\begin{align}
\nabla \cdot\left(h^{-1}\nabla \psi\right)+f&=c_0 h \psi + c_1 h \tag{1}\\
g\nabla h &= \frac{f}{h}\nabla\psi \tag{2}
\end{align}
$$

where $f$ is a known function of $y$, and $g$, $c_0$, $c_1$ are all constants.  Is it possible to solve this PDE set using `py-pde`, with proper boundary conditions?

**Comment by david-zwicker** [comment]
Unfortunately, I don't think `py-pde` is the right package to solve these equations. Since it is based on finite-differences, it cannot cope with arbitrary boundary conditions easily. I recommend looking into finite element or finite volume methods.

**Reply by miniufo** [reply]
Thanks for the quick reply.  How about in a cartesian x-y plane?  Is it possible to solve Eqs. (1-2) on a rectangular domain?

**Reply by david-zwicker** [reply]
This is still challenging since these are not time-dependent PDEs. I guess you could re-formulate the PDEs by introducing some fictitious relaxation dynamics, so the stationary state solution corresponds to solution of your equations.
